mods/ts/weapons/energyweapons.yaml

```diff
@@ -39,17 +77,43 @@ MechRailgun:
 		DamageTypes: Prone100Percent, TriggerProne, FireDeath
 
 SonicZap:
-	ReloadDelay: 120
+	ReloadDelay: 180
 	Range: 6c0
 	Charges: yes
 	Report: SONIC4.AUD
-	Projectile: LaserZap
-		Color: 200,0,255,255
-		BeamWidth: 12
-		BeamDuration: 50
+	Projectile: LineImpactProjectile
+		Speed: 0c64
+		TicksBetweenImpacts: 5 # Roughly 18 impacts.
+		BeyondTargetRange: 0c256
+		MaxTargetRange: 50c0
+		MinRangeSource: 0c0
+		FalloffMaxReduction: 0
+		FalloffStartDistance: 49c0
+		FalloffEndDistance: 50c0
+		BeamWidth: 0c256
+		Blockable: true
+		BeamRenderWidth: 384
+		BeamFireDuration: 90
+		Color: 200,0,40,255
 	Warhead@1Dam: SpreadDamage
-		Spread: 42
-		Damage: 100
+		Range: 0, 32
+		Falloff: 100, 100
+		Damage: 8
+		InfDeath: 6
+		AffectsParent: false
+		ValidStances: Neutral, Enemy
+		Versus:
+			Heavy: 80
+			Concrete: 60
+		DamageTypes: Prone50Percent, TriggerProne, FireDeath
+	Warhead@2Dam: SpreadDamage
+		Range: 0, 32
+		Falloff: 50, 50
+		Damage: 8
+		InfDeath: 6
+		InvalidTargets: Disruptor # Does not affect friendly disruptors at all
+		AffectsParent: false
+		ValidStances: Ally
```

Comment:
Are you referring to the
```Versus:
 			Heavy: 80
 			Concrete: 60``` ?
They should be in. At least they appear to be for me.
The idea of this part (nearly exact copy-paste) is simply to half the damage done in friendly fire, and do no damage at all to friendly disruptors.
Otherwise disruptors quickly become unusable in armies.